component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/emailotp/EmailOTPAuthenticator.java

```diff
@@ -114,40 +136,78 @@ public boolean canHandle(HttpServletRequest request) {
-        } else if (StringUtils.isNotEmpty(request.getParameter(EmailOTPAuthenticatorConstants.EMAIL_ADDRESS))) {
+        } else if (StringUtils.isNotEmpty(request.getParameter(EMAIL_ADDRESS))) {
             // if the request comes with EMAIL ADDRESS, it will go through this flow.
             initiateAuthenticationRequest(request, response, context);
             return AuthenticatorFlowStatus.INCOMPLETE;
-        } else if (StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.CODE)) &&
-                StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.RESEND))) {
-            // if the request comes with code, it will go through this flow.
-            initiateAuthenticationRequest(request, response, context);
+        } else if (StringUtils.isEmpty(request.getParameter(CODE)) && StringUtils.isEmpty(request.getParameter(RESEND)))
+        {
+            // if the request doesn't contain an OTP code and resend is not set, it will go through this flow
+            if (!isIDFInitiatedFromEmailOTP(context)) {
+                // if the IDF is not initiated from within EmailOTP authenticator (this decides the state of the flow)
+                if (context.getLastAuthenticatedUser() == null) {
+                    // this is the case that no prior authentication step has been done, and IDF has not been triggered
+                    context.setProperty(WITHOUT_AUTHENTICATED_USER, true);
+                    redirectUserToIDF(response, context); // Redirect to username entering page
+                    context.setProperty(IS_IDENTIFIER_FIRST_INITIATED_FROM_AUTHENTICATOR, true);
+                    return AuthenticatorFlowStatus.INCOMPLETE;
+                }
+            }
+            if (context.getLastAuthenticatedUser() == null) {
+                // when username is obtained through IDF page and the user is not yet set in the context
+                AuthenticatedUser user = resolveUser(request, context);
+                setResolvedUserInContext(context, user);
+            }
+            try {
+                /* if the username is invalid, this will throw an exception and the user will be redirected to OTP page.
+                This will generate and prepare the email OTP to send */
+                initiateAuthenticationRequest(request, response, context);
+            } catch (AuthenticationFailedException e) {
+                if (context.getProperty(USER_NAME) == null) { // null checker for the username
+                    if (Boolean.parseBoolean(IdentityUtil.getProperty(NOTIFY_USER_EXISTENCE))) {
+                        // The case where user has entered an invalid username
+                        log.error(e + ": The entered user is not in the user stores.");
+                    }
+                    Map<String, String> emailOTPParameters = getAuthenticatorConfig().getParameterMap();
+                    try {
+                        // redirect the user to OTP entering page
```

Comment:
```suggestion
                        // Redirect the user to OTP entering page.
```
Starts the comments with a capital letter and end with a full stop.
Even the old comments has a different format, follow this in the comments that you add

Reply:
Resolved. Thanks!